### DISCUSSION TITLE
Object oriented without a GUI-IDE?

### DISCUSSION THREAD

**Discussion by richmondmathewson** [elan-language/LanguageAndIDE#732] [Ideas]
If Elan really is object oriented why does it look like a crappy version of the BASIC and FORTRAN I studied in 1975 (I am 62), and why does it NOT have a modern drag-and-drop GUI-IDE like, for instance, OXT Lite?

https://openxtalk.org/
![SShot 2024-09-26 at 14 08 12](https://github.com/user-attachments/assets/5e5a462f-3606-4b9f-b197-a410a9c827f2)

**Comment by richardpawson** [comment]
An object-oriented _language_ and an object-oriented UI are two entirely different things.  Many of the early OOUIs were written in entirely procedural languages (principally C at that time). And many OO languages have had no GUI at all associated with them. Smalltalk was the exception. Alan Kay, whom I have known personally for 30+ years very generously wrote the forward to my school-level textbook introducing OOP.

Elan is written for a very specific purpose (teaching programming in schools), and has a very clear set of design goals. You would do well to do a little more research, or ask a few more open questions, before launching into criticism.

 P.S. I was programming in Fortran before you ;-)
